phoenix-core-server/src/main/java/org/apache/phoenix/iterate/NonAggregateRegionScannerFactory.java

```diff
@@ -298,6 +303,24 @@ static OrderedResultIterator deserializeFromScan(Scan scan, RegionScanner s,
         }
     }
 
+    private static class IteratorAndContext {
```

Comment:
@palashc you think this is what seems tricky? Let me take a look next week in detail.

Reply:
@virajjasani Yes, exactly. Maybe there is another way to refactor here.

Reply:
I did not understand what we are achieving here. OrderedResultIterator is used in both client and server side. So, we need to add the next() with ScannerContext method to OrderedResultIterator too. This method will be used on the server side. The scanner context should not serialized/deserialized. We also need to pass through ScannerContext context for the other result iterators used on the server side (for example, OffsetResultIterator). Also let us add tests for these result iterators.

Reply:
As discussed offline, the changes in this class help with linking the RegionScanner from NonAggregateRegionScannerFactory to the RegionScannerResultIterator, thus bypassing the ResultIterators in between since the current codebase structure makes it difficult to pass the ScannerContext through them on the server side. 
I have also renamed the inner class used here to more meaningful `OrderedResultIteratorWithScannerContext`